jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1264,8 +1264,13 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
         return managers;
     }
 
+    /**
+     * @deprecated use {@link SslContextFactory.Server#newSniX509ExtendedKeyManager(X509ExtendedKeyManager)} instead
+     */
+    @Deprecated
     protected X509ExtendedKeyManager newSniX509ExtendedKeyManager(X509ExtendedKeyManager keyManager)
     {
+        // Will throw a NPE.
         return new SniX509ExtendedKeyManager(keyManager);
```

Comment:
This actually creates an instance that is unusable, so let's throw `UnsupportedOperationException` from here, and only override on the `Server` subclass.